vllm/config/model.py

```diff
@@ -1752,6 +1757,114 @@ def get_and_verify_max_len(self, max_model_len: int):
         logger.info("Using max model len %s", max_model_len)
         return max_model_len
 
+    @property
+    def attn_type(self) -> AttnTypeStr:
+        if self.pooler_config is not None:
+            pooling_type = self._model_info.default_pooling_type.lower()
+            if pooling_type == "cls":
+                return "encoder_only"
+            else:
+                is_causal = getattr(self.hf_config, "is_causal", True)
+                return "encoder_only" if not is_causal else self._model_info.attn_type
+        elif self.is_hybrid:
+            return "hybrid"
+        elif self.is_attention_free:
+            return "attention_free"
+        elif self.is_encoder_decoder:
+            return "encoder_decoder"
+        else:
+            return "decoder"
+
+    @property
+    def is_chunked_prefill_supported(self) -> bool:
+        attn_type = self.attn_type
+        if self.pooler_config is not None:
+            # for pooling models
+            if attn_type == "encoder_only":
+                logger.debug(
+                    "Pooling models with bidirectional attn does not support "
+                    "chunked prefill."
+                )
+                return False
+            else:
+                # for decoder models (causal attn)
+                pooling_type = self.pooler_config.pooling_type.lower()
+                if pooling_type == "all":
+                    logger.debug(
+                        "Pooling models with mean pooling does not support "
+                        "chunked prefill."
+                    )
+                    return False
+                elif pooling_type == "mean":
+                    logger.debug(
+                        "Pooling models with all pooling does not support "
+                        "chunked prefill."
+                    )
+                    return False
```

Comment:
1. The logging and comments are missed up. Please clean it up. 

For example, `pooling_type == 'all'` should have a logging.debug message of "Pooling models with all pooling...", same for `pooling_type == 'mean'`.

2. The comment "# for decoder models (causal attn)" before `pooling_type == 'all'` doesn't seem appropriate there. No?